docs/design.md

```diff
@@ -6,35 +6,27 @@ Parca Agent uses BPF CO-RE (Compile Once – Run Everywhere) using [libbpf](http
 
 From a high-level it performs the following steps:
 
-* [Target Discovery](#target-discovery): Discovers cgroups to attach the profiler on the machine the Parca Agent is running on.
+* [Target Discovery](#target-discovery): Discovers Kubernetes clusters, if any, and fetches its metadata.
```

Comment:
I think this should be called something else.
Our targets now are all the processes on the system.

We should add a step to describe how we provide context for the processes using container runtimes, cgroups and systems units.

Reply:
"Metadata Discovery"?

Reply:
Both sound great. You pick one :)